lib/MongoDB/Op/_Command.pm

```diff
@@ -87,9 +87,19 @@ sub execute {
     # $query is passed as a reference because it *may* be replaced
     $self->_apply_read_prefs( $link, $topology_type, $self->{query_flags}, \$self->{query});
 
-    my ( $op_bson, $request_id ) =
-      MongoDB::_Protocol::write_query( $self->{db_name} . '.$cmd',
-        $self->{bson_codec}->encode_one( $self->{query} ), undef, 0, -1, $self->{query_flags});
+    my ( $op_bson, $request_id );
+
+    if ( $ENV{DO_OP_MSG} ) {#$link->supports_op_msg ) {
+        push @{$self->{query}}, ( '$db', $self->db_name );
+        my @sections = MongoDB::_Protocol::prepare_sections( $self->{bson_codec}, $self->{query} );
+        $self->{query} = \@sections;
```

Comment:
I'm not sure why you're replacing this here.  If you leave it alone, then command monitoring should still do the right thing and you don't have to reassemble it.  Think about it this way: OP_MSG is just a way of *laying out* the query in bytes -- the actual query doesn't change (except for addition of metadata like $db and read preference.).

Reply:
The issue is being able to test that OP_MSG specifically has been used - the test plan says to ensure each document is done as a document sequence properly. Also the command-monitoring spec states that `When a driver sends an OP_MSG with a document sequence, it MUST include the document sequence as a BSON array in CommandStartedEvent.command.`

Reply:
I think you're reading the command monitoring spec backwards.  It's saying that in the case of OP_MSG -- which might split a field out of the command into a sequence -- then command monitoring must present it reassembled, so that apps consuming events see them recombined.